save-backend/src/main/kotlin/org/cqfn/save/backend/repository/TimestampBasedFileSystemRepository.kt

```diff
@@ -104,6 +114,56 @@ class TimestampBasedFileSystemRepository(configProperties: ConfigProperties) {
             }
     }
 
+    /**
+     * @param owner owner name
+     * @param part file part
+     * @return Mono with number of bytes saved
+     * @throws FileAlreadyExistsException if file with this name already exists
+     */
+    fun saveImage(part: Mono<FilePart>, owner: String): Mono<ImageInfo> = part.flatMap { part ->
+        val uploadedDir = rootDirImage.resolve(owner)
+        val organization = organizationRepository.findByName(owner)
+
+        uploadedDir.apply {
+            if (exists()) {
+                listDirectoryEntries().forEach { it.deleteIfExists() }
+            }
+        }.deleteIfExists()
+
+        uploadedDir
+            .createDirectories()
+            .resolve(part.filename()).run {
+                createFile(this, part)
+                    .collect(Collectors.summingLong { it })
+                    .map {
+                        logger.info("Saved $it bytes into $this")
+                        val relativePath = ("/$owner/$name")
+                        organization.avatar = relativePath
+                        organizationRepository.save(organization)
```

Comment:
And this should be performed in organizationService as well, as a handler of successful upload

Reply:
Fixed